### PR TITLE
Update UG

### PR DESCRIPTION
**Pull request by ExtraShotLatte**
Added improvements to User Guide (UG) based on CS2101's comments!
- Added target audience for UG.
- Fixed weird language used for optimization of Paidlancers for fast typist.
- Refactor saving data to save data for title of subtopic
- Added note for user to know that opening Paidlancers adds in extra files
- Added small tip on what Command Prompt is
- Added more explanation for commands
- Changed `home folder` to `main folder` for better understanding in UG

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -425,8 +454,8 @@ Paidlancers data are saved in the hard disk automatically on command issue. Ther
 
 **Q**: How do I transfer my data to another Computer?
 1. Follow the quickstart guide until step 3.
```
Comment (glozxi) [inline docs/UserGuide.md:456]: It might be good to hyperlink the quickstart guide.

**Review by glozxi — CHANGES_REQUESTED** [review]
I like the friendly tone!

I've made some suggestions that you can take a look at. There are also some nits that I have spotted from previous PRs' that you may want to fix:

1. In the `edit` section, `Edits will replace existing values, edits are not cumulative.` and `Existing tags will also be removed when editing editing tags, it will be necessary to input all the existing tags during the edit.` seem to be saying the same thing. Additionally, `Existing tags will also be removed when editing editing tags, it will be necessary to input all the existing tags during the edit.` has an extra `editing`.

3. `Saving data` header and in features summary could be changed to `Save data` to be more consistent with the other features.

Overall, I like how you improved the UG 🧇

File: docs/UserGuide.md [hunk 2]
```diff
@@ -298,7 +319,9 @@ Finds event(s) from the event book based on the given search string.
 
 ### Link Contact to Event: `linkcontact` <a id = "link-contact-to-event"></a>
 
-Links a client contact to an event.
+This command will links a client contact to an event.
```
[the remark this answers is not in the document]
Comment (ExtraShotLatte) [reply docs/UserGuide.md:322]: Good catch! Fixed!

**Comment by ExtraShotLatte** [comment]
@glozxi Updated the UG with your inputs! Kindly take another look at this! :)

Have also updated the initial PR comment to include changes that are in this PR for better reference!

**Comment by ExtraShotLatte** [comment]
@AY2223S2-CS2103T-T11-3/developers Latest update deployed on my branch [here](https://extrashotlatte.github.io/tp/UserGuide.html) for you guys' perusal!

**Review by glozxi — APPROVED** [review]
LGTM 👍

**Pull request merged**